roofit/roofitcore/inc/RooAddModel.h

```diff
@@ -16,6 +16,7 @@
 #ifndef ROO_ADD_MODEL
 #define ROO_ADD_MODEL
 
+#include "RooAddPdf.h"
```

Comment:
Is it not possible to forward declare the RooAddPdf dependency ?

Reply:
No, it is not because it depends on a the `RooAbsPdf::CacheElem`, and it is not possible to do a nested class forward declaration.

I have therefore followed your suggestion to move the common functionality into a new file.